### PR TITLE
feat: [PIE-6053]: Allowed Values support for KubernetesInfraSpec

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Review by ishantmehtaharness — DISMISSED** [review]
Looks Good
this is so simple to use. Thanks to Chetan and Ramesh

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx [hunk 1]
```diff
@@ -191,6 +203,7 @@ const KubernetesInfraSpecEditable: React.FC<KubernetesInfraSpecEditableProps> =
                       formik.setFieldValue('namespace', value)
                     }}
                     isReadonly={readonly}
+                    allowedValuesType={ALLOWED_VALUES_TYPE.NUMBER}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:206]: Is this number field ? namespace is not number field. Please remove this change.

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx [hunk 2]
```diff
@@ -228,6 +241,7 @@ const KubernetesInfraSpecEditable: React.FC<KubernetesInfraSpecEditableProps> =
                             formik.setFieldValue('releaseName', value)
                           }}
                           isReadonly={readonly}
+                          allowedValuesType={ALLOWED_VALUES_TYPE.NUMBER}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:244]: Same as above. Remove this change.

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx [hunk 3]
```diff
@@ -289,6 +303,10 @@ const KubernetesInfraSpecInputForm: React.FC<KubernetesInfraSpecEditableProps &
             className={css.connectorMargin}
             multiTypeProps={{ allowableTypes, expressions }}
             enableConfigureOptions={false}
+            templateProps={{
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:306]: I think this change is required in Runtime view, not in Edit view. Please get in touch with Ramesh if you need more explanation.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx (hunk 3 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:306]: My bad. I think this change is fine.

**Review by rameshx — APPROVED** [review]

**Comment by pankajsingh-harness** [comment]
retrigger jest

**Comment by pankajsingh-harness** [comment]
retrigger jest

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx [hunk 4]
```diff
@@ -203,7 +203,7 @@ const KubernetesInfraSpecEditable: React.FC<KubernetesInfraSpecEditableProps> =
                       formik.setFieldValue('namespace', value)
                     }}
                     isReadonly={readonly}
-                    allowedValuesType={ALLOWED_VALUES_TYPE.NUMBER}
+                    allowedValuesType={ALLOWED_VALUES_TYPE.TEXT}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:206]: You do not need to make any change here. We are making changes for Number and ConnectorRef field and this field is neither of them.

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx [hunk 5]
```diff
@@ -241,7 +241,7 @@ const KubernetesInfraSpecEditable: React.FC<KubernetesInfraSpecEditableProps> =
                             formik.setFieldValue('releaseName', value)
                           }}
                           isReadonly={readonly}
-                          allowedValuesType={ALLOWED_VALUES_TYPE.NUMBER}
+                          allowedValuesType={ALLOWED_VALUES_TYPE.TEXT}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:244]: Same as above.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx (hunk 4 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/KubernetesInfraSpec/KubernetesInfraSpec.tsx:206]: This is related to last time changes we made for the number field. This field may have been missed last time. I found out this so updated. Do I need to create separate PR for this?

**Review by chetanvekariya-harness — DISMISSED** [review]

**Comment by pankajsingh-harness** [comment]
retrigger jest

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**